build_tools/build_detect_platform

```diff
@@ -134,7 +134,7 @@ case "$TARGET_OS" in
         ;;
     FreeBSD)
         PLATFORM=OS_FREEBSD
-        COMMON_FLAGS="$COMMON_FLAGS -fno-builtin-memcmp -D_REENTRANT -DOS_FREEBSD"
+        COMMON_FLAGS="$COMMON_FLAGS -fno-builtin-memcmp -D_REENTRANT -DOS_FREEBSD -D_GLIBCXX_USE_C99"
```

Comment:
@adamretter I removed `-fpermissive` and it does not actually affect the build (those warnings still treated as errors).
I've built it with `gmake rocksdbjava DISABLE_WARNING_AS_ERROR=1`